pkgs/games/unvanquished/default.nix

```diff
@@ -0,0 +1,139 @@
+{ lib, stdenv, writeScript, fetchzip, fetchurl, fetchgit, buildFHSUserEnv, runCommandNoCC, makeDesktopItem
+, gcc, cmake, gmp, libGL, zlib, ncurses , geoip, lua5, nettle
+, curl, SDL2, freetype, glew , openal, libopus, opusfile, libogg
+, libvorbis, libjpeg, libwebp, libpng
+, libtheora # TODO: drop once we get to 0.52
+, cacert, aria2 # to download assets
+}:
+
+let version = "0.51.2";
+    binary-deps-version = "4";
+
+    unvanquished-binary-deps = stdenv.mkDerivation rec {
+      # DISCLAIMER: this is selected binary crap from the NaCl SDK
+      pname = "unvanquished-binary-deps";
+      version = binary-deps-version;
+      src = fetchzip {
+        url = "https://dl.unvanquished.net/deps/linux64-${version}.tar.bz2";
+        sha256 = "08bpyavbh5lmyprvqqi59gnm8s1fjmlk9f1785wlv7f52d9f9z1p";
+      };
+      dontPatchELF = true;
+      preFixup = ''
+        # We are not using the autoPatchelfHook, because it would make
+        # nacl_bootstrap_helper unable to load nacl_loader:
+        # "nacl_loader: ELF file has unreasonable e_phnum=13"
+        interpreter="$(< "$NIX_CC/nix-support/dynamic-linker")"
+        for f in pnacl/bin/*; do
+          if [ -f "$f" && -x "$f" ]; then
+            echo "Patching $f"
+            patchelf --set-interpreter "$interpreter" "$f"
+          fi
+        done
+      '';
+      preCheck = "pnacl/bin/clang -v"; # check it links correctly
+      meta = {
+        platform = "x86_64-linux";
+      };
+      installPhase = ''
+        mkdir -p $out
+        cp -R ./* $out/
+      '';
+    };
+
+    preload = stdenv.mkDerivation {
+      name = "nix-preload-for-unvanquished-nacl";
+      buildCommand = ''
+        mkdir $out/etc -p
+        echo ${gcc.cc.lib}/lib/libstdc++.so.6 > $out/etc/ld-nix.so.preload
+      '';
+    };
+
+    fhsEnv = buildFHSUserEnv {
+      name = "unvanquished-fhs-wrapper";
+      targetPkgs = pkgs: [ gcc.cc.lib preload ];
+    };
+
+    unvanquished-assets = stdenv.mkDerivation {
+      name = "unvanquished-assets";
+      inherit version;
+      src = fetchurl {
+        # master as of 2021-04-09: the 0.51.2 script isn't as great
+        url = "https://raw.githubusercontent.com/Unvanquished/Unvanquished/2783e77acc2e41fc1f717124394ed851f1b612e6/download-paks";
+        sha256 = "1zbsi0yzfvgassg6dqks66qpjqa4yxi2ymis2qh8aj9dvkgqk6im";
+      };
+      outputHash = "sha256:15vndprms0vi5rbvf67hqxwgs3xhlcrxbzn4z3kz9d54g0z1zbzh";
+      outputHashMode = "recursive";
+      nativeBuildInputs = [ aria2 cacert ];
+      meta = {
+        license = lib.licenses.cc-by-sa-25;
+        description = "Assets for the Unvanquished game";
+      };
+      buildCommand = "bash $src .";
+      installPhase = "cp ./* $out/";
+    };
+
+# this really is the daemon game engine, the game itself is in the assets
+in stdenv.mkDerivation rec {
+  name = "unvanquished-${version}";
+  inherit version;
+  inherit binary-deps-version;
+  src = fetchgit {
```

Comment:
Please use fetchFromGitHub.

Reply:
I can't really use fetchFromGitHub here, because there are more than 5 submodules that would need to be handled manually otherwise.

Reply:
There is also a `fetchSubmodules` parameter in `fetchFromGithHub`, which turns `fetchFromGitHub` into **surprise** fetchgit. Also it makes no difference code-wise I think it would still make sense to switch to it since becomes more obvious why it is required. Right now the motivation for using fetchgit is not clear and the same question would arise in future.

Reply:
Also if GitHub would ever change something about their API or introduce a better way to download repos with submodules we could introduce/fix that for all packages that use fetchFromGitHub instead of doing it for every package.

Reply:
updated